pkg/controllers/allocation/scheduling/scheduler.go

```diff
@@ -147,3 +156,56 @@ func IsSchedulable(pod *v1.Pod, node *v1.Node) bool {
 	// TODO, support node affinity
 	return true
 }
+
+type schedulerMetricsDecorator struct {
+	scheduler                Scheduler
```

Comment:
Optional: 

type metricsDecoratedScheduler struct {
  Scheduler // embedded
  ...
}

Reply:
Tried it.  The go compiler won't allow embedding an interface within a struct.

The compiler will allow embedding the `scheduler` struct but that had the effect of coupling `metricsDecoratedScheduler` to `scheduler`, rather than allowing `metricsDecoratedScheduler` to work with any `Scheduler`.

Reply:
I think of the metricsdecorator as a subclass of scheduler (with is ~= to embedding in golang, though not quite the same). I tend to see the go philosophy as leaning away from things like interfaces, setters, getters, and instead towards duck typing and direct variable access. I also expect that we won't have multiple scheduler implementations, but I'm fine either way.